src/main/scala/devices/debug/Debug.scala

```diff
@@ -644,16 +646,23 @@ class TLDebugModuleOuterAsync(device: Device)(implicit p: Parameters) extends La
     })
     val rf_reset = IO(Input(Reset()))
 
-    dmi2tlOpt.foreach { _.module.io.dmi <> io.dmi.get }
+    childClock := io.dmi_clock
+    childReset := io.dmi_reset
 
-    dmiBypass.module.io.bypass := ~io.ctrl.dmactive | ~AsyncResetSynchronizerShiftReg(in=io.ctrl.dmactiveAck, sync=3, name=Some("dmactiveAckSync"))
+    withClockAndReset(childClock, childReset) {
+      dmOuter.module.clock := io.dmi_clock
+      dmOuter.module.reset := io.dmi_reset
 
-    io.ctrl <> dmOuter.module.io.ctrl
-    io.innerCtrl <> ToAsyncBundle(dmOuter.module.io.innerCtrl, AsyncQueueParams.singleton(safe=cfg.crossingHasSafeReset))
-    dmOuter.module.io.hgDebugInt := io.hgDebugInt
-    io.hartResetReq.foreach { x => dmOuter.module.io.hartResetReq.foreach {y => x := y}}
-    io.dmAuthenticated.foreach { x => dmOuter.module.io.dmAuthenticated.foreach { y => y := x}}
+      dmi2tlOpt.foreach { _.module.io.dmi <> io.dmi.get }
+
+      dmiBypass.module.io.bypass := ~io.ctrl.dmactive | ~AsyncResetSynchronizerShiftReg(in=io.ctrl.dmactiveAck, sync=3, name=Some("dmactiveAckSync"))
```

Comment:
but I think you are right you need to put these non-lazy modules in explicit withClockAndReset blocks, but actually I am not sure. @jackkoenig , @terpstra do you know if this would automatically get the `childClock` and `childReset` domain without a withClockAndReset(childClock, childReset)?

Reply:
Yes, without the withClockAndReset, there is a FIRRTL exception.